zulipterminal/ui_tools/boxes.py

```diff
@@ -572,6 +586,35 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                     self.view.controller.save_draft_confirmation_popup(
                         this_draft,
                     )
+        elif is_primary_direction_key(key):
+            header = self.header_write_box
+            if self.to_write_box is None:
+                if (self.focus_position == self.FOCUS_CONTAINER_HEADER
+                        and header.focus_col == self.FOCUS_HEADER_BOX_STREAM):
+                    cursor_pos = header[self.FOCUS_HEADER_BOX_STREAM].edit_pos
+                    stream_name = (header[self.FOCUS_HEADER_BOX_STREAM]
+                                   .edit_text)
+                    stream_name_len = len(stream_name)
+                    if (not self.model.is_valid_stream(stream_name)
+                            and key in ['right', 'down']):
+                        if key != 'right' or cursor_pos == stream_name_len:
+                            self.footer_notify_invalid_stream()
+                            return key
+            elif key == 'down':
+                if self.focus_position == self.FOCUS_HEADER_BOX_RECIPIENT:
+                    self.update_recipient_emails(self.to_write_box)
+                    invalid_emails = self.model.get_invalid_recipient_emails(
+                        self.recipient_emails
+                    )
+                    if invalid_emails:
+                        self.footer_notify_invalid_emails(invalid_emails)
+                        return key
```

Comment:
This part of the code seems the same as the section validating on cycling out using `tab`, can you perhaps add a helper within `WriteBox` to do this?

Reply:
Sure, I was thinking of adding a helper but then I hesitated, I don't know why. :)

Reply:
It will be difficult to return the `key`(incase of invalid email) from inside a helper function?

Reply:
You could move the part above the `invalid_emails` to a helper (would be a prep for), but I guess it wouldn't be very necessary here since it's just a direct call to the model's method. I'll add the helper in #937 directly then.